scripts/babel/rewrite-modules.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+function mapModule(context, module) {
+  // Look in options passed to babel for `_moduleMap` which contains a map of
+  // replacements. This allows other packages to be published to npm and used
+  // directly without being a part of the same build.
+  var moduleMap = context.state.opts._moduleMap || {};
+  if (moduleMap.hasOwnProperty(module)) {
+    return moduleMap[module];
+  }
+  // Jest understands the haste module system, so leave modules intact.
+  if (process.env.NODE_ENV !== 'test') {
+    return './' + module;
+  }
+}
+
+module.exports = function(babel) {
+  var t = babel.types;
+
+  /**
+   * Transforms `require('Foo')` and `require.requireActual('Foo')`.
+   */
+  function transformRequireCall(context, call) {
+    if (
+      !t.isIdentifier(call.callee, {name: 'require'}) &&
+      !(
+        t.isMemberExpression(call.callee) &&
+        t.isIdentifier(call.callee.object, {name: 'require'}) &&
+        t.isIdentifier(call.callee.property, {name: 'requireActual'})
+      )
+    ) {
+      return;
+    }
+    var moduleArg = call.arguments[0];
+    if (moduleArg && moduleArg.type === 'Literal') {
+      var module = mapModule(context, moduleArg.value);
+      if (module) {
+        return t.callExpression(
```

Comment:
We can re-use the MemberExpression for `require.requireActual`.

This is different from Jest because the `jest` methods can be chained. If chained, the `member.object` has to be point at a new CallExpression. (However, like here, we're still able to re-use the `jest` identifier.)